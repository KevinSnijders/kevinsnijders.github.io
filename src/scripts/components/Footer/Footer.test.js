import React from 'react';
import {
  setUpComponent,
  testComponentMatchSnapshot,
  testComponentIsRendered,
  findByHTML
} from '../../utils/index';
import Footer from './Footer';

const testComponent = 'FooterComponent';
const state = {
  socials: [
    {
      icon: 'github-square-brands',
      link: '#'
    },
    {
      icon: 'linkedin-brands',
      link: '#'
    }
  ]
};

describe('Footer Component', () => {
  describe('Match Snapshot', () => {
    testComponentMatchSnapshot(Footer, state);
  });
  describe('Check PropTypes', () => {});
  describe('Footer correct state', () => {
    let component;
    beforeEach(() => {
      component = setUpComponent(Footer, state);
    });
    it('Should render without errors', () => {
      testComponentIsRendered(component, testComponent, true);
    });
  });
  describe('Social icons', () => {
    let component;
    beforeEach(() => {
      component = setUpComponent(Footer);
      component.setState({
        socials: []
      });
    });
    it('Should not render socials without state', () => {
      testComponentIsRendered(component, 'IconComponent', false);
    });
  });
});
