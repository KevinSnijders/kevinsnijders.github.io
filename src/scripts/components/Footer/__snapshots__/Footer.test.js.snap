// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<footer
  className="footer"
  data-test="FooterComponent"
  id="contact"
>
  <div
    className="col-sm-auto fade-in-bottom"
  >
    <a
      className="btn btn--light mb-4"
      data-test="ButtonComponent"
      href="#"
    >
      Contact me
    </a>
    <ul
      className="d-flex flex-row justify-content-center mb-4"
      data-test="SocialComponent"
    >
      <li
        className="pl-2 pr-2"
      >
        <a
          href="#"
        >
          <svg
            className="icon icon-github-square-brands "
            data-test="IconComponent"
          >
            <use
              xlinkHref="#.github-square-brands"
            />
          </svg>
        </a>
      </li>
      <li
        className="pl-2 pr-2"
      >
        <a
          href="#"
        >
          <svg
            className="icon icon-linkedin-brands "
            data-test="IconComponent"
          >
            <use
              xlinkHref="#.linkedin-brands"
            />
          </svg>
        </a>
      </li>
    </ul>
    <p
      className="footer__copyright"
    >
      Kevin Snijders Â© 
      2019
       all rights reserved
    </p>
    <div
      className="footer__back-to-top"
    >
      <a
        href="#top"
      >
        <svg
          className="icon icon-keyboard_arrow_up bounce"
          data-test="IconComponent"
        >
          <use
            xlinkHref="#.keyboard_arrow_up"
          />
        </svg>
      </a>
    </div>
  </div>
</footer>
`;
